### PR TITLE
feat: Config.exclude can be array

### LINKED ISSUE #558: Config.exclude: can be array
### Describe the feature

For more fine grained control, we should be able to pass a string array `exclude` field in `houdini.config.js`.

Maybe `include` also ?

Due to my project structure, I need to pass multiple globs to exclude, like I'm able to do in `graphqlrc.yaml`.

Thank you!

### Criticality

nice to have

### PR DESCRIPTION
**Pull request by DanielHritcu**
# In progress

Fixes #558 

Allow passing an array of strings to `Config.exclude`. Also allow string for this to be a non breaking change.

Should `Config.include` support arrays as well ?

### To help everyone out, please make sure your PR does the following:

- [x] Update the first line to point to the ticket that this PR fixes
- [x] Add a message that clearly describes the fix
- [ ] If applicable, add a test that would fail without this fix
- [x] Make sure the unit and integration tests pass locally with `pnpm run tests` and `cd integration && pnpm run tests`
- [x] Includes a changeset if your fix affects the user with `pnpm changeset`

### CODE REVIEW

File: src/common/config.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export class Config {
 	schemaPath?: string
 	persistedQueryPath?: string
 	include: string
```
Comment (AlecAivazis) [inline src/common/config.ts:31]: Let's make this same change to include too 👍

**Review by AlecAivazis — CHANGES_REQUESTED** [review]
Thanks for submitting this @DanielHritcu! Mind updating the config docs page too?  It lives in [site/src/routes/api/config.svx](https://github.com/HoudiniGraphql/houdini/blob/main/site/src/routes/api/config.svx).

File: src/vite/index.ts [hunk 2]
```diff
@@ -42,7 +42,10 @@ export default function ({
 					}
 
 					// make sure that the file doesn't match the exclude
-					return !config.exclude || !minimatch(filepath, config.exclude)
+					return (
```
Comment (AlecAivazis) [inline src/vite/index.ts:45]: I realize this was there before, but I think we should try to use `config.includeFile` here if possible

File: src/common/config.ts (hunk 1 above)
Comment (DanielHritcu) [reply src/common/config.ts:31]: Will do.

**Comment by DanielHritcu** [comment]
I added the code for include, but I can't manage to build the integration project to test.

I get this weird error, and can't figure out why it's thrown.

The files get included, and the artifacts generated.

Can be replicated starting here: https://github.com/HoudiniGraphql/houdini/pull/559/commits/f149302df95a2b08277a7ea580ca4979ebc4581d

The error I get:
```
file: /Users/danielhritcu/GitHub/houdini/integration/$houdini/runtime/lib/index.js:5:9
3: export { errorsToGraphQLLayout } from './errors';
4: export * from './types';
5: export * as log from './log';
            ^
6: export * from './deepEquals';
7: // putting this here was the only way i could find to reliably avoid import issues
[WebServer] error during build:
Error: Parse error @:5:10
    at parse$b (file:///Users/danielhritcu/GitHub/houdini/node_modules/.pnpm/vite@3.1.0/node_modules/vite/dist/node/chunks/dep-665b0112.js:33612:355)
    at Object.transform (file:///Users/danielhritcu/GitHub/houdini/node_modules/.pnpm/vite@3.1.0/node_modules/vite/dist/node/chunks/dep-665b0112.js:42799:27)
Error: Process from config.webServer was not able to start. Exit code: 1
```

**Comment by AlecAivazis** [comment]
@DanielHritcu are you still running into that error?

**Comment by DanielHritcu** [comment]
@AlecAivazis yes I am.

**Comment by AlecAivazis** [comment]
Oh, interesting it's happening on github too - alright so it _must_ be related to your changes since these tests ran successfully in your other PR. I'll try to figure out what's going on

**Comment by AlecAivazis** [comment]
@DanielHritcu I pushed up a few changes that I think have fixed the problem. I'm really not sure how it's any different from what you have but i'm not getting the errors any more. Let me know what you think

**Review by AlecAivazis — APPROVED** [review]

**Pull request merged**